File: leetCode/heeseok/study/selectSort.py
class selectSort:
    list = []

    # constructor
    # def __init__(self, nums:int):
    #     self.list = nums

    def change(self,list:list, x, y):
        tmp = list[x]
        self.list[x] = self.list[y]
        self.list[y] = tmp

    def execute(self, nums: list):
        self.list = nums
        listSize = len(self.list)

        if listSize > 1:
            for i in range(0, listSize-1):
                minNum = self.list[i]
                minIndex = i
                for j in range(i+1, listSize):
                    if minNum > self.list[j]:
                        minNum = self.list[j]
                        minIndex = j
                self.change(self.list,i,minIndex)
                print(i+1,"번째-->",self.list)

        return self.list


arr = [6, 1, 7, 8, 9, 3, 5, 4, 2]
print(0,"번째-->",arr)
print("result-->",selectSort().execute(arr))

File: leetCode/yeon/SelectSort.java
class Solution {
    public int[] sortArray(int[] nums) {
        
        int tmpIndex ;
        int tmp ;
        
        for(int i = 0; i < nums.length ; i++){
            tmpIndex = i;
            tmp = nums[i];
            for(int j = i ; j < nums.length; j++ ){
                // 제일 작은수 찾기
                if( tmp > nums[j]){
                    tmpIndex = j;
                    tmp = nums[j];
                }
            }
            
            nums[tmpIndex] = nums[i];
            nums[i] = tmp;
        }
        
        return nums;      
            
    }
}
